e2e/rbd.go

```diff
@@ -539,14 +539,26 @@ var _ = Describe("RBD", func() {
 					LabelSelector: fmt.Sprintf("app=%s", app.Name),
 				}
 				filePath := app.Spec.Containers[0].VolumeMounts[0].MountPath + "/test"
-				_, stdErr, err = execCommandInPod(
-					f,
-					fmt.Sprintf("echo 'Hello World' > %s", filePath),
-					app.Namespace,
-					&appOpt)
-				if err != nil || stdErr != "" {
-					e2elog.Failf("failed to write IO, err: %v, stdErr: %v ", err, stdErr)
-				}
+				err = wait.PollImmediate(poll, timeout, func() (bool, error) {
+					_, stdErr, err = execCommandInPod(
+						f,
+						fmt.Sprintf("echo 'Hello World' > %s", filePath),
+						app.Namespace,
+						&appOpt)
+					if err != nil {
+						e2elog.Failf("failed to run write command: %v", err)
```

Comment:
this will cause the test to fail, can it happen that `err` is non-nil, and the filesystem or block-device is read-only?

Reply:
I think I need to reorder this check:

I need to check for stdErr first and then if that is equal to "Read-only ..." then retry, if stdErr != "Read-only" then err should bailout.

Thanks, I think I recorded at the last moment.